example-app/server.js

```diff
@@ -64,19 +64,71 @@ socketServer.of("chat").on("login", function(username) {
 	userLogout(username);
 });
 
+// serves up a browser-compatible version of req.URL (which should correspond to a .JS file) by // using modConCat. Cache output, and only regenerate whenever original .JS file changes
+function concatAndCache(baselocation, req, res) {
+
+	var inputFile	= baselocation + req.url;
+	var outputFile 	= inputFile + ".modConcat";
+		
+	// compare dates on input vs. output files
+	// if input file is newer, we need to regenerate output file
+	// otherwise, just serve up output file
+	
+	fs.stat(inputFile,  (err, stat_input) => {
+		fs.stat(outputFile, (err, stat_output) => {
+
+			var NeedsRegenerated;
+		
+			// input file doesn't exist?? nothing we can do
+			if (stat_input === undefined) return;	
+			
+			// output file doesn't exist?? then we definitely need to regenerate it
+			if (stat_output === undefined) 
```

Comment:
Suggest using `== null` instead of `=== undefined`.  Also I'd recommend always using curly braces after if / else statements, even though it's not strictly necessary.